users/views.py

```diff
@@ -1,14 +1,101 @@
-from django.shortcuts import render, redirect
-from django.contrib.auth.forms import UserCreationForm
-
-# Create your views here.
-def register(request):
-    if request.method == 'POST':
-        form  = UserCreationForm(request.POST)
-        if form.is_valid():
-            form.save()
-            username = form.cleaned_data.get('username')
-            return redirect('http://localhost:8000/accounts/login')
-    else:
-        form  = UserCreationForm()
-    return render(request, 'users/register.html', {'form': form})
+from .models import User, Wallet
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework.authtoken.models import Token
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.permissions import IsAuthenticated
+from .serializers import (RegisterSerializer, ChangePasswordSerializer,
+                          LoginSerializer, ProfileSerializer,
+                          BalanceSerializer)
+
+
+class LoginView(APIView):
+    def post(self, request):
+        serializer = LoginSerializer(data=request.data)
+        if serializer.is_valid():
+            email = serializer.validated_data["username"]
+            password = serializer.validated_data["password"]
+            try:
+                user = User.objects.get(email=email)
+                if user.check_password(password):
+                    token = Token.objects.get(user=user).key
+                    return Response(
+                        f'User logged-in.\
```

Comment:
common error handler missing, also error object should be consistent json object